systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMakerST.java

```diff
@@ -663,6 +666,36 @@ void testScaleMirrorMakerSubresource() {
             assertThat(pod.contains(mmGenName), is(true));
         }
     }
+
+    @Test
+    void testScaleMirrorMakerToZero() {
+        // Deploy source kafka
+        KafkaResource.kafkaEphemeral(kafkaClusterSourceName, 1, 1).done();
+        // Deploy target kafka
+        KafkaResource.kafkaEphemeral(kafkaClusterTargetName, 1, 1).done();
+
+        KafkaMirrorMakerResource.kafkaMirrorMaker(CLUSTER_NAME, kafkaClusterTargetName, kafkaClusterSourceName, "my-group" + rng.nextInt(Integer.MAX_VALUE), 3, false)
+            .editMetadata()
+                .addToLabels("type", "kafka-mirror-maker")
+            .endMetadata()
+            .done();
+
+        String mmDepName = KafkaMirrorMakerResources.deploymentName(CLUSTER_NAME);
+        List<String> mmPods = kubeClient().listPodNames("type", "kafka-mirror-maker");
+        assertThat(mmPods.size(), is(3));
+
+        LOGGER.info("Scaling MirrorMaker to zero");
+        KafkaMirrorMakerResource.replaceMirrorMakerResource(CLUSTER_NAME, mm -> mm.getSpec().setReplicas(0));
+
+        KafkaMirrorMakerUtils.waitForKafkaMirrorMakerReady(CLUSTER_NAME);
```

Comment:
How does this ensure that the previous status is not caught by this? I do not see any checks for observed generations etc.

Reply:
Gonna add it, thanks

Reply:
https://github.com/strimzi/strimzi-kafka-operator/pull/3438/files#diff-22f8df9b640e22843eae57444b9be687R700

Reply:
Check is added

Reply:
So should this line be removed now?